include/deal.II/base/tensor.h

```diff
@@ -2028,7 +2005,8 @@ template <int rank_1,
           int rank_2,
           int dim,
           typename Number,
-          typename OtherNumber>
+          typename OtherNumber,
+          typename = typename std::enable_if<rank_1 >= 0 && rank_2 >= 0>::type>
```

Comment:
Ugh...
```suggestion
          typename = typename std::enable_if<(rank_1 >= 0 && rank_2)>= 0>::type>
```
But is this enough? You're returning a `Tensor<rank_1+rank_2-2, dim>`. I think the condition should actually be
```suggestion
          typename = typename std::enable_if<(rank_1 + rank_2 >= 2)>::type>
```

Reply:
You are right. The condition actually has to be `rank_1 >= 1 && rank_2 >= 1` to ensure that the ranks of the arguments and the result are all nonnegative. I have updated the statement accordingly.

Reply:
`rank_1 + rank_2 >= 2` is also incomplete because 0 + 2 = 2, or -1 + 3 = 2 :smile: